### ISSUE TITLE
YAML MetaBlocks should be interpreted as list of Paras, not Paras + Plain

### ISSUE THREAD

**Issue by msprev** [jgm/pandoc#5271]
```
---
abstract: |
  This is the abstract.

  It consists of two paragraphs.
...
```

The content of the abstract field should be interpreted ([pandoc manual](https://pandoc.org/MANUAL.html#extension-yaml_metadata_block)) as:

```
<p>This is the abstract.</p>
<p>It consists of two paragraphs.</p>
```

But it actually is interpreted as:

```
<p>This is the abstract.</p>
It consists of two paragraphs.
```

In other words, the last element of the MetaBlocks field is interpreted as `Plain` rather than `Para`. This looks like undesirable behaviour?

---

Note that the final element can be 'forced' to be Para by adding an extra line after it that contains >=3 whitespaces (any less won't work):

```
---
abstract: |
  This is the abstract.

  It consists of two paragraphs.
___
...
```

```
<p>This is the abstract.</p>
<p>It consists of two paragraphs.</p>
```

But this seems unintuitive and it is easy to lose invisible characters on that line, which will revert by to the old behaviour.

**Comment by tarleb** [comment]
I'd argue that the current behavior is correct, but that the documentation should be updated.

Note that you can get the desired result by using

``` yaml
---
abstract: |+
  This is the abstract.

  It consists of two paragraphs.

___
```

Note the `|+`, which ensures that trailing newlines are kept.

**Comment by mb21** [comment]
Then again, the markdown reader also appends a newline to its input:

    ~ echo 'foo' | pandoc -t native   
    [Para [Str "foo"]]

**Comment by tarleb** [comment]
Fair point. So should we append a single newline to those blocks?  It should be possible to recover the current behavior by using `|-` instead of `|` (just in case someone *expects* Plain values).

**Comment by jgm** [comment]
This seems to be a regression. In pandoc 1.19.2.1 (just tested), we get two Paras. That is the intended behavior.  Starting with 2.0 perhaps (tested 2.0.6) we get Para and Plain -- not intended.  Not sure what changed to cause this, but we should definitely fix it.
